### PR TITLE
EOS-13993: NFS ADDB: ADDB for all file operations

### PR DESCRIPTION
**Pull request by vimalraghwani20**
**Problem Statement**
EOS-13993: NFS ADDB: ADDB for all file operations

**Problem Description**
Add ADDB traces for all file operations

**Solution Overview**
Added ADDB traces for following file operations

- open
- close
- status
- reopen
- commit

**Companion PRs**
https://github.com/Seagate/cortx-fs-ganesha/pull/43
https://github.com/Seagate/cortx-utils/pull/68

**UT**
Updated at https://github.com/Seagate/cortx-utils/pull/68

### CODE REVIEW

File: src/cortxfs/cortxfs_fops.c [hunk 1]
```diff
@@ -115,6 +120,7 @@ int cfs_creat_ex(struct cfs_fs *cfs_fs, cfs_cred_t *cred, cfs_ino_t *parent,
 		(void) cfs_unlink(cfs_fs, cred, parent, &object, name);
 		(void) kvs_discard_transaction(kvstor, &index);
 	}
+	perfc_trace_finii(PERFC_TLS_POP_DONT_VERIFY);
```
Comment (pratyush-seagate) [inline src/cortxfs/cortxfs_fops.c:123]: Should we trace rc?

File: src/cortxfs/cortxfs_fops.c [hunk 2]
```diff
@@ -72,6 +73,9 @@ int cfs_creat(struct cfs_fs *cfs_fs, cfs_cred_t *cred, cfs_ino_t *parent_ino,
 
 	log_trace("parent_ino=%llu name=%s newfile_ino=%llu rc=%d",
 		  *parent_ino, name, *newfile_ino, rc);
+	perfc_trace_attr(PEA_CFS_CREATE_PARENT_INODE, *parent_ino);
+	perfc_trace_attr(PEA_CFS_NEW_FILE_INODE, *newfile_ino);
+	perfc_trace_finii(PERFC_TLS_POP_DONT_VERIFY);
 	return rc;
```
Comment (pratyush-seagate) [inline src/cortxfs/cortxfs_fops.c:79]: Should we trace rc?

Comment (vimalraghwani20) [reply src/cortxfs/cortxfs_fops.c:80]: fixed

File: src/cortxfs/cortxfs_fops.c (hunk 1 above)
Comment (vimalraghwani20) [reply src/cortxfs/cortxfs_fops.c]: fixed

**Review by AtitaShirwaikar — APPROVED** [review]
+1

File: src/cortxfs/cortxfs_fops.c [hunk 3]
```diff
@@ -72,6 +73,10 @@ int cfs_creat(struct cfs_fs *cfs_fs, cfs_cred_t *cred, cfs_ino_t *parent_ino,
 
 	log_trace("parent_ino=%llu name=%s newfile_ino=%llu rc=%d",
 		  *parent_ino, name, *newfile_ino, rc);
+	perfc_trace_attr(PEA_CFS_CREATE_PARENT_INODE, *parent_ino);
+	perfc_trace_attr(PEA_CFS_NEW_FILE_INODE, *newfile_ino);
+	perfc_trace_finii(PEA_CFS_RES_RC);
```
Comment (pratyush-seagate) [inline src/cortxfs/cortxfs_fops.c:78]: What does it mean? Are you trying to use perfc_trace_finii as perfc_trace_attr?

File: src/cortxfs/cortxfs_fops.c [hunk 4]
```diff
@@ -115,6 +121,8 @@ int cfs_creat_ex(struct cfs_fs *cfs_fs, cfs_cred_t *cred, cfs_ino_t *parent,
 		(void) cfs_unlink(cfs_fs, cred, parent, &object, name);
 		(void) kvs_discard_transaction(kvstor, &index);
 	}
+	perfc_trace_finii(PEA_CFS_RES_RC);
```
Comment (pratyush-seagate) [inline src/cortxfs/cortxfs_fops.c:124]: What does it mean? Are you trying to use perfc_trace_finii as perfc_trace_attr?

File: src/cortxfs/cortxfs_fops.c (hunk 3 above)
Comment (vimalraghwani20) [reply src/cortxfs/cortxfs_fops.c]: fixed.



-bash-4.2$ ls -lrt / | grep addb
drwx------    3 root root     4096 Sep 29 00:50 addb_25843
drwx------    3 root root     4096 Sep 29 00:50 addb_25920
drwx------    3 root root     4096 Oct 19 01:59 addb_1010
drwx------    3 root root     4096 Oct 19 01:59 addb_1088
drwx------    3 root root     4096 Oct 19 02:19 addb_15253
drwx------    3 root root     4096 Oct 19 02:19 addb_15328
drwx------    3 root root     4096 Oct 29 09:14 addb_25868
drwx------    3 root root     4096 Oct 29 09:14 addb_25943
drwx------    3 root root     4096 Oct 29 22:14 addb_22928
drwx------    3 root root     4096 Oct 29 22:14 addb_23001
drwx------    3 root root     4096 Nov  3 00:09 addb_24435
drwx------    3 root root     4096 Nov  3 00:09 addb_24510
drwx------    3 root root     4096 Nov  3 00:23 addb_7900
drwx------    3 root root     4096 Nov  3 00:23 addb_7973
-bash-4.2$ sudo m0addb2dump -p /usr/lib64/libcortx-utils.so -- /addb_7973/o/100000000000000\:2 | grep fsuser > addb_dump_op_decode_6
-bash-4.2$ cat addb_dump_op_decode_6 | grep cfs_result_rc
* 2020-11-03-00:25:02.239037385 fsuser_attribute cfs_create, attribute,  64b,        cfs_result_rc, 0
* 2020-11-03-00:25:02.245876292 fsuser_attribute cfs_create_ex, attribute, 64a,      cfs_result_rc, 0
* 2020-11-03-00:25:02.281896665 fsuser_attribute cfs_create, attribute,  780,        cfs_result_rc, 0
* 2020-11-03-00:25:02.287954425 fsuser_attribute cfs_create_ex, attribute, 77f,      cfs_result_rc, 0
* 2020-11-03-00:25:02.367043898 fsuser_attribute cfs_create, attribute,  9ce,        cfs_result_rc, 0
* 2020-11-03-00:25:02.374253897 fsuser_attribute cfs_create_ex, attribute, 9cd,      cfs_result_rc, 0
* 2020-11-03-00:25:02.410446855 fsuser_attribute cfs_create, attribute,  af5,        cfs_result_rc, 0
* 2020-11-03-00:25:02.417331338 fsuser_attribute cfs_create_ex, attribute, af4,      cfs_result_rc, 0

File: src/cortxfs/cortxfs_fops.c (hunk 4 above)
Comment (vimalraghwani20) [reply src/cortxfs/cortxfs_fops.c]: Fixed.

-bash-4.2$ ls -lrt / | grep addb
drwx------    3 root root     4096 Sep 29 00:50 addb_25843
drwx------    3 root root     4096 Sep 29 00:50 addb_25920
drwx------    3 root root     4096 Oct 19 01:59 addb_1010
drwx------    3 root root     4096 Oct 19 01:59 addb_1088
drwx------    3 root root     4096 Oct 19 02:19 addb_15253
drwx------    3 root root     4096 Oct 19 02:19 addb_15328
drwx------    3 root root     4096 Oct 29 09:14 addb_25868
drwx------    3 root root     4096 Oct 29 09:14 addb_25943
drwx------    3 root root     4096 Oct 29 22:14 addb_22928
drwx------    3 root root     4096 Oct 29 22:14 addb_23001
drwx------    3 root root     4096 Nov  3 00:09 addb_24435
drwx------    3 root root     4096 Nov  3 00:09 addb_24510
drwx------    3 root root     4096 Nov  3 00:23 addb_7900
drwx------    3 root root     4096 Nov  3 00:23 addb_7973
-bash-4.2$ sudo m0addb2dump -p /usr/lib64/libcortx-utils.so -- /addb_7973/o/100000000000000\:2 | grep fsuser > addb_dump_op_decode_6
-bash-4.2$ cat addb_dump_op_decode_6 | grep cfs_result_rc
* 2020-11-03-00:25:02.239037385 fsuser_attribute cfs_create, attribute,  64b,        cfs_result_rc, 0
* 2020-11-03-00:25:02.245876292 fsuser_attribute cfs_create_ex, attribute, 64a,      cfs_result_rc, 0
* 2020-11-03-00:25:02.281896665 fsuser_attribute cfs_create, attribute,  780,        cfs_result_rc, 0
* 2020-11-03-00:25:02.287954425 fsuser_attribute cfs_create_ex, attribute, 77f,      cfs_result_rc, 0
* 2020-11-03-00:25:02.367043898 fsuser_attribute cfs_create, attribute,  9ce,        cfs_result_rc, 0
* 2020-11-03-00:25:02.374253897 fsuser_attribute cfs_create_ex, attribute, 9cd,      cfs_result_rc, 0
* 2020-11-03-00:25:02.410446855 fsuser_attribute cfs_create, attribute,  af5,        cfs_result_rc, 0
* 2020-11-03-00:25:02.417331338 fsuser_attribute cfs_create_ex, attribute, af4,      cfs_result_rc, 0

**Review by pratyush-seagate — APPROVED** [review]
LGTM, +1

**Review by sachinpunadikar — APPROVED** [review]
+1

**Pull request merged**